purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesFactory.kt

```diff
@@ -127,6 +139,25 @@ internal class PurchasesFactory(
         return checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED
     }
 
+    private fun createDiagnosticsManagerIfNeeded(
+        context: Context,
+        backend: Backend,
+        dispatcher: Dispatcher,
+        diagnosticsEnabled: Boolean
+    ): DiagnosticsManager? {
+        return if (diagnosticsEnabled) {
+            DiagnosticsManager(
+                DiagnosticsFileHelper(FileHelper(context)),
```

Comment:
Well, `PurchasesFactory` is a hard class to mock, since it's where all our real dependencies actually get instantiated. We could try to refactor this class to better separate all these dependencies and allow us to test the dependency instantiation process we do in this class, but that seems out of the scope of this PR.